File: sudhdkso/42626.cpp
#include <string>
#include <vector>
#include<queue>
#include<iostream>
using namespace std;

int solution(vector<int> scoville, int K) {
    int answer = 0,sum=0;
    priority_queue<int, vector<int>, greater<int>>pq(scoville.begin(),scoville.end());

    while (pq.size()>1 && pq.top()<K) {
        sum = pq.top(); pq.pop(); sum += (pq.top()*2); pq.pop();
        pq.push(sum);
        answer++;

    }

    if (pq.top() < K) return -1;

    return answer;
}

File: qwertycvb/42626.py
import heapq


def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)

    while scoville[0] < K:
        heapq.heappush(scoville, heapq.heappop(scoville) + heapq.heappop(scoville) * 2)
        answer += 1
        if len(scoville) == 1 and scoville[0] < K:
            return -1

    return answer